xrpl/models/amounts/mpt_amount.py

```diff
@@ -24,7 +24,7 @@ class MPTAmount(BaseModel):
     :meta hide-value:
     """
 
-    value: Union[str, int, float] = REQUIRED  # type: ignore
+    value: str = REQUIRED  # type: ignore
```

Comment:
I don't think this should be changed at the xrpl package level. Doesn't binary-codec take `int` or `float` and eventually convert it to string? cc @mvadari

Reply:
I didn't review the MPT code - here it is, for reference: https://github.com/XRPLF/xrpl-py/blob/main/xrpl/core/binarycodec/types/amount.py#L304

Reply:
1. We have discrepancies in handling the translation of rippled's `std::uint64_t` type into a standard JSON type. For instance, `asset_price` (in `OracleSet` transaction) accepts an `int` whereas the MPTAmount.value field accepts `str` only. Mea culpa, I implemented the `PriceOracle` amendment. 

Is it acceptable to use `str` JSON type for all the `std::uint64_t` fields? Do we need to provide any int, float interfaces for these `sf<Fieldname>` values?

2. As shown in test cases inside `test_amount.py` file, we need to support scientific notation syntax in the `MPTAmount.value` fields. At the moment, scientific notation does not work for serialization.

Reply:
@khancode 

https://github.com/XRPLF/xrpl-py/blob/13fdbd8003ddf2aa60e398f8063fed51d138e020/xrpl/core/binarycodec/types/amount.py#L128

It is not possible to do a string-based search for `.` character in a `float, int` input. Do MPT amounts not allow for fractional portion? 

It is possible to extend this check for int, float types by processing it through the `Decimal` library. We can extract the `exponent` portion of the number and determine the validity of the value.

This was one of the differences I found between serialization of IOU and MPT values. I'd prefer to retain the existing functionality instead of reducing the API surface area (in terms of allowed types for `value` field). What do you think ?

Reply:
@ckeshava You can verify on the docs if it accepts it. And if you have an idea for a solution, you can verify it by testing your changes.

Reply:
I'll address that in a separate PR. I've created an issue to track it: https://github.com/XRPLF/xrpl-py/issues/816